app/models/secret_storage.rb

```diff
@@ -50,6 +50,43 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(SECRET_BACKEND  + self.encode(key))
+      result.data[:secret_data]
+    end
+
+    def self.write(key, data)
```

Comment:
data is a hash with lots of things that are not secret ... feels strange to dump everything into secrets-data ... won't be usable via the cli / from kubernetes after